src/Microsoft.AspNet.OData.Shared/Query/ODataQueryOptions.cs

```diff
@@ -426,11 +441,19 @@ public virtual IQueryable ApplyTo(IQueryable query, ODataQuerySettings querySett
                 if (resultsLimited && InternalRequest.RequestUri != null && InternalRequest.RequestUri.IsAbsoluteUri &&
                     InternalRequest.Context.NextLink == null)
                 {
-                    Uri nextPageLink = InternalRequest.GetNextPageLink(pageSize);
-                    InternalRequest.Context.NextLink = nextPageLink;
+                    DefaultQuerySettings settings = Context.DefaultQuerySettings;
+                    if (settings.EnableSkipToken)
+                    {
+                        InternalRequest.Context.PageSize = pageSize;
```

Comment:
Now that we are exposing a PageSize property, and we know it's value, any reason not to set it even for the non-skip token case?  I.e., you could do:

InternalRequest.Context.PageSize = pageSize;
if (!settings.EnableSkipToken)
{
  InternalRequest.Context.NextLink = InternalRequest.GetNextPageLink(pageSize);
}